src/Data/Product/N-ary/Heterogeneous.agda

```diff
@@ -313,32 +313,33 @@ infix 5 Π[_]
 ------------------------------------------------------------------------
 -- n-ary pointwise liftings
 
+lift₂ : ∀ n {ls r s t} {as : Sets n ls} {R : Set r} {S : Set s} {T : Set t} →
+        (R → S → T) → as ⇉ R → as ⇉ S → as ⇉ T
+lift₂ zero    op f g = op f g
+lift₂ (suc n) op f g = λ x → lift₂ n op (f x) (g x)
+
```

Comment:
This can also be written

```agda
lift₂ n op f g = curryₙ n (λ vs → op (uncurryₙ n f vs) (uncurryₙ n g vs))
```

but not convinced it's worth the trouble.

Reply:
I have n-ary lift too but the type is quite nasty: https://github.com/gallais/nary/blob/3541bd0eac93eadc0998ae477c64bdf17b6703e4/doc/lagda/N-ary.lagda#L532

May be worth including once the work is split up in more granular modules.

Reply:
included the general lift during the refactoring